src/data/resolvers/coinMachine.ts

```diff
@@ -262,6 +262,49 @@ export const coinMachineResolvers = ({
         return null;
       }
     },
+    async coinMachineTotalTokens(_, { colonyAddress }) {
+      try {
+        const coinMachineClient = await colonyManager.getClient(
+          ClientType.CoinMachineClient,
+          colonyAddress,
+        );
+        const tokenClient = (await colonyManager.getClient(
+          ClientType.TokenClient,
+          colonyAddress,
+        )) as TokenClient;
+
+        const currentTokenBalance = await coinMachineClient.getTokenBalance();
+
+        /*
+         * We use the `FromChain` suffix to make these events more easily
+         * recognizable when reading the code
+         */
+        const transferLogsFromChain = await getLogs(
+          tokenClient,
+          tokenClient.filters.Transfer(null, coinMachineClient.address, null),
+        );
```

Comment:
This is very expensive, but I guess it's what we have to do in order to fetch the token transfers into the coin machine.

However, you can filter it even more, by only fetching blocks from the time the current instance of the coin machine was installed.

See this for an example:
https://github.com/JoinColony/colonyDapp/blob/9ba1bbf9606d3afd7ee897ce2ae5fdee1eb75f71/src/data/resolvers/coinMachine.ts#L329-L356

Reply:
Fair enough - done now.